spec/controllers/collection_participants_controller_spec.rb

```diff
@@ -0,0 +1,464 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe CollectionParticipantsController do
+  include LoginMacros
+
+  def it_redirects_to_with_notice(path, notice)
+    it_redirects_to(path)
+    expect(flash[:notice]).to eq notice
+  end
+
+  def it_redirects_to_with_error(path, error)
+    it_redirects_to(path)
+    expect(flash[:error]).to eq error
+  end
+
+  def it_redirects_to(path)
```

Comment:
Amusingly, I have added exactly this same method in one of my PRs too (copied from your previous PR, I think). However, I think this means we're going to use this quite a lot, so it might make sense to either put it at the end of the `spec_helper`, or maybe create another module like `LoginHelpers` for it with your notice and error variations above, so they can all be reused everywhere that tests controller actions with redirects.